lib/topics/state_and_behaviour.rb

```diff
@@ -2,4 +2,27 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # StateAndBehaviour::Car
+  class Car
+    attr_accessor :year, :color, :model, :current_speed
+
+    def self.default_car
+      new {}
+    end
+
+    def initialize(attributes)
+      self.year          = attributes.fetch(:year, 2000)
+      self.color         = attributes.fetch(:color, 'green')
+      self.model         = attributes.fetch(:model, 'skyline')
+      self.current_speed = 0
+    end
+
+    def speed_up(speed)
+      self.current_speed += speed
+    end
+
+    def push_break(speed)
+      self.current_speed -= speed unless self.current_speed <= speed
```

Comment:
hard readable line try ti return from method unless this condition
and then only execute what you need